RecoLocalTracker/SiPhase2VectorHitBuilder/interface/VectorHitMomentumHelper.h

```diff
@@ -0,0 +1,25 @@
+#ifndef VectorHitMomentumHelper_H
+#define VectorHitMomentumHelper_H
+
+#include "DataFormats/TrackerRecHit2D/interface/VectorHit.h"
+#include "MagneticField/Engine/interface/MagneticField.h"
+
+class VectorHitMomentumHelper {
+public:
+  VectorHitMomentumHelper();
+  ~VectorHitMomentumHelper() {
+  }
+
+  float transverseMomentum(VectorHit& vh,const MagneticField* magField){
+	  GlobalPoint center(0.0, 0.0, 0.0);
+	  float magnT = magField->inTesla(center).mag();
+	  double rho = 1. / vh.curvatureORphi("curvature").first;
+	  //0.003 is because the curvature (rho) is in cm and not in m
+	   return (0.003 * magnT * rho);
```

Comment:
what 0.003 is????

Reply:
That is from the standard pT [GeV] = 0.3 * B[T] * R [m], taking into account that rho is in cm, so you get 0.003.

Reply:
ok is speedOfLight: 2.99792458e-3F (note: float)
yes https://cmssdt.cern.ch/lxr/source/MagneticField/Engine/src/MagneticField.cc#0019
actually never realized that was in a int ....
in any case mag() is wrong: should be z() anyhow...

Reply:
I am now using this value and switched from mag() to z()